### ISSUE TITLE
no_std support for sapio modules?

### ISSUE THREAD

**Issue by JeremyRubin** [sapio-lang/sapio#243]
it would be nice if we could guarantee sapio modules can work with no std

**Comment by ProofOfKeags** [comment]
What does this buy us IRL? And iirc nostd implies no access to an allocator, is that going to be an issue?

**Comment by JeremyRubin** [comment]
no, you can have an allocator with nostd, you just need to enable the crate (see https://docs.rust-embedded.org/book/intro/no-std.html)

https://www.reddit.com/r/rust/comments/9eyc21/noob_what_exactly_is_no_std_and_why_is_it_so/ has a good answer

> no_std exists because std assumes you have an operating system with things like a memory allocator and a file system. This assumption is not true on things like microcontrollers, or when you are the operating system. Rust automatically links std by default, so there has to be a switch to turn it off, or Rust wouldn't be usable in those places.

we don't have access to filesystem APIs, etc, so nostd would avoid things that could be problematic
